### PR TITLE
issue-544, refactor externalChanges message

### PR DESCRIPTION
**Pull request by OleksiienkoMykyta**

### CODE REVIEW

File: controllers/clusters/helpers.go [hunk 1]
```diff
@@ -158,8 +158,36 @@ func getSortedAppVersions(versions []*models.AppVersions, appType string) []*ver
...
 }
 
+func removeRedundantFieldsFromSpec(k8sSpec interface{}, ignoreFields ...string) (string, error) {
+	k8sSpecJson, err := json.Marshal(k8sSpec)
+	if err != nil {
+		return "", err
+	}
+
+	if len(ignoreFields) == 0 {
+		return string(k8sSpecJson), nil
```
Comment (ribaraka) [inline controllers/clusters/helpers.go:168]: You can just return bytes

Comment (OleksiienkoMykyta) [reply controllers/clusters/helpers.go]: fixed

File: controllers/clusters/helpers.go [hunk 2]
```diff
@@ -158,8 +158,36 @@ func getSortedAppVersions(versions []*models.AppVersions, appType string) []*ver
 	return nil
 }
 
+func removeRedundantFieldsFromSpec(k8sSpec interface{}, ignoreFields ...string) (string, error) {
```
[the remark this answers is not in the document]
Comment (OleksiienkoMykyta) [reply controllers/clusters/helpers.go]: fixed

File: controllers/clusters/helpers.go [hunk 3]
```diff
@@ -158,8 +158,36 @@ func getSortedAppVersions(versions []*models.AppVersions, appType string) []*ver
...
+	if err != nil {
+		return "", err
+	}
+
+	if len(ignoreFields) == 0 {
+		return string(k8sSpecJson), nil
+	}
+
+	k8sSpecMap := map[string]interface{}{}
+	err = json.Unmarshal([]byte(string(k8sSpecJson)), &k8sSpecMap)
```
[the remark this answers is not in the document]
Comment (OleksiienkoMykyta) [reply controllers/clusters/helpers.go]: fixed

File: controllers/clusters/helpers.go [hunk 4]
```diff
@@ -158,8 +158,36 @@ func getSortedAppVersions(versions []*models.AppVersions, appType string) []*ver
...
+	k8sSpecJson, err := json.Marshal(k8sSpec)
+	if err != nil {
+		return "", err
+	}
+
+	if len(ignoreFields) == 0 {
+		return string(k8sSpecJson), nil
+	}
+
+	k8sSpecMap := map[string]interface{}{}
```
[the remark this answers is not in the document]
Comment (OleksiienkoMykyta) [reply controllers/clusters/helpers.go]: fixed

**Review by ribaraka — APPROVED** [review]

**Review by testisnullus — APPROVED** [review]

**Pull request merged**